tests/lib/uc20-create-partitions/main.go

```diff
@@ -72,22 +87,42 @@ func main() {
 		panic(err)
 	}
 
-	var model *asserts.Model
-	if args.ModelPath != "" {
-		var err error
-		model, err = readModel(args.ModelPath)
-		if err != nil {
-			panic(fmt.Sprintf("cannot load model: %v", err))
-		}
-	}
+	obs := &simpleObserver{}
+
 	options := install.Options{
-		Mount:      args.Mount,
-		Encrypt:    args.Encrypt,
-		KernelPath: args.KernelPath,
-		Model:      model,
+		Mount:   args.Mount,
+		Encrypt: args.Encrypt,
 	}
-	err = installRun(args.Positional.GadgetRoot, args.Positional.Device, options, nil)
+	err = installRun(args.Positional.GadgetRoot, args.Positional.Device, options, obs)
 	if err != nil {
 		panic(err)
 	}
+
+	if args.Encrypt {
+		loadChain := []bootloader.BootFile{
```

Comment:
Hm, it feels like this test becomes a bit artificial with this change. I may be wrong, I don't know the code as well as you. My feeling is that in the real installRun() code we have an observer that does the sealing(?) so I wonder if we should follow a similar pattern here? i.e. use this code in the sealer or even call the right helper (sealKeyToModeenv() ?) here? But maybe I'm a) totally off b) this happens in followups anyway.

Reply:
I wondered the same and my question here is whether we should keep the original test scope or just test encryption without sealing since that's what gadget/install does now. I was inclined in this direction, however I kept the scope to avoid removing tests before discussing it.